### PR TITLE
[v2.9] Fix delete init node for RKE2 clusters

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

Rancher PR for shepherd PR https://github.com/rancher/shepherd/pull/181.

### CODE REVIEW

File: go.mod
Comment (slickwarren) [inline go.mod:15]: reminder to leave in draft state if there's a replace like this in the go.mod

File: tests/v2/validation/deleting/delete_init_machine_test.go
Comment (slickwarren) [inline tests/v2/validation/deleting/delete_init_machine_test.go:55]: imo the original way was more concise

File: tests/v2/validation/deleting/delete_init_machine_test.go
Comment (slickwarren) [inline tests/v2/validation/deleting/delete_init_machine_test.go:56]: does this run on rke1 now?

**Review by slickwarren — COMMENTED** [review]
some questions about the removal of some lines

File: tests/v2/validation/deleting/delete_init_machine_test.go
Comment (markusewalker) [reply tests/v2/validation/deleting/delete_init_machine_test.go:56]: It doesn't, was this intentional? I was just generally cleaning, I can put it back though.

File: tests/v2/validation/deleting/delete_init_machine_test.go
Comment (markusewalker) [reply tests/v2/validation/deleting/delete_init_machine_test.go:55]: I don't have a preference, I can put it back. Just following the practice in the other files.

Comment (markusewalker) [reply tests/v2/validation/deleting/delete_init_machine_test.go]: Resolved.

File: tests/v2/validation/deleting/delete_init_machine_test.go
Comment (markusewalker) [reply tests/v2/validation/deleting/delete_init_machine_test.go]: Resolved.

**Review by slickwarren — COMMENTED** [review]
lgtm: pending shepherd merge for final approval

**Review by susesgartner — APPROVED** [review]
LGTM

**Review by slickwarren — APPROVED** [review]

**Review by Josh-Diamond — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]

**Pull request merged**
